compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala

```diff
@@ -79,7 +79,7 @@ object SyntaxHighlighting {
         } else {
           if (n.isUpper && (keywordStart || prev == '.')) {
             appendWhile(n, !typeEnders.contains(_), typeDef)
-          } else if (keywordStart) {
+          } else if (keywordStart || prev.isDigit) {
```

Comment:
Shouldn't you fix `keywordStart` instead?

Reply:
I was thinking about it too and maybe I should have asked first. 
The other place where ```keywordStart``` is used is type ascriptions (the ```if``` just before the one I modified). And as far as I know there is no situation where a number literal can precede a type ascription. For this reason I only added it here. 
But maybe it would be better to add it to ```keywordStart```.